api/Dockerfile

```diff
@@ -0,0 +1,51 @@
+# elixir version and erlang version are defined in .tool_versions
+ARG ELIXIR_VERSION=1.14.4
+ARG ERLANG_VERSION=25.3
+
+# this is the latest as of 27 april 2023
+ARG ALPINE_VERSION=3.17.2
+
+# use this image to install dependencies and build the application
+FROM hexpm/elixir:${ELIXIR_VERSION}-erlang-${ERLANG_VERSION}-alpine-${ALPINE_VERSION} as builder
+
+# create a directory and add the project files to it
+WORKDIR /root
+
+# define this as a prod environment
+ENV MIX_ENV=prod
+
+COPY assets assets
+COPY config config
+COPY lib lib
+COPY mix.* /root/
+
+#  install hex, rebar, and dependencies
+RUN mix local.hex --force
+RUN mix local.rebar --force
+RUN mix deps.get --only prod
+
+# build the application
+RUN mix release 
+
+# use this image to run the application
+FROM alpine:${ALPINE_VERSION}
```

Comment:
i switched to a multi stage build as suggested in the new app checklist.

after looking at some of the other projects, this seems to be the pattern they are using.